core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -3667,45 +3668,58 @@ public void setLastBlockSeenHeight(int lastBlockSeenHeight) {
         }
     }
 
-    public void setLastBlockSeenTimeSecs(long timeSecs) {
+    public void setLastBlockSeenTime(Instant time) {
         lock.lock();
         try {
-            lastBlockSeenTimeSecs = timeSecs;
+            lastBlockSeenTime = checkNotNull(time);
         } finally {
             lock.unlock();
         }
     }
 
+    public void clearLastBlockSeenTime() {
+        lock.lock();
+        try {
+            lastBlockSeenTime = null;
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    /** @deprecated use {@link #setLastBlockSeenTime(Instant)} or {@link #clearLastBlockSeenTime()} */
+    @Deprecated
+    public void setLastBlockSeenTimeSecs(long timeSecs) {
+        checkArgument(timeSecs > 0);
+        setLastBlockSeenTime(Instant.ofEpochSecond(timeSecs));
+    }
+
     /**
-     * Returns the UNIX time in seconds since the epoch extracted from the last best seen block header. This timestamp
+     * Returns time extracted from the last best seen block header, or empty. This timestamp
      * is <b>not</b> the local time at which the block was first observed by this application but rather what the block
      * (i.e. miner) self declares. It is allowed to have some significant drift from the real time at which the block
      * was found, although most miners do use accurate times. If this wallet is old and does not have a recorded
      * time then this method returns zero.
      */
-    public long getLastBlockSeenTimeSecs() {
+    public Optional<Instant> getLastBlockSeenTimeInstant() {
         lock.lock();
         try {
-            return lastBlockSeenTimeSecs;
+            return Optional.ofNullable(lastBlockSeenTime);
         } finally {
             lock.unlock();
         }
     }
 
-    /**
-     * Returns a {@link Date} representing the time extracted from the last best seen block header. This timestamp
-     * is <b>not</b> the local time at which the block was first observed by this application but rather what the block
-     * (i.e. miner) self declares. It is allowed to have some significant drift from the real time at which the block
-     * was found, although most miners do use accurate times. If this wallet is old and does not have a recorded
-     * time then this method returns null.
-     */
+    /** @deprecated use {@link #getLastBlockSeenTimeInstant()} */
+    @Deprecated
+    public long getLastBlockSeenTimeSecs() {
+        return getLastBlockSeenTimeInstant().map(time -> time.getEpochSecond()).orElse((long) 0);
+    }
+
+    /** @deprecated use {@link #getLastBlockSeenTimeInstant()} */
+    @Deprecated
     @Nullable
     public Date getLastBlockSeenTime() {
-        final long secs = getLastBlockSeenTimeSecs();
-        if (secs == 0)
-            return null;
-        else
-            return new Date(secs * 1000);
+        return getLastBlockSeenTimeInstant().map(time -> Date.from(time)).orElse(null);
```

Comment:
`Date::from`

Reply:
Doesn't work: "Object is not a functional interface"

Reply:
Works – thanks!

Reply:
It doesn't look like this change actually made it into the PR...